internal/schemas/gen/gen.go

```diff
@@ -81,13 +85,22 @@ func gen() error {
 
 	log.Println("ensuring terraform is installed")
 
-	tmpDir, err := ioutil.TempDir("", "tfinstall")
+	installDir, err := ioutil.TempDir("", "hcinstall")
 	if err != nil {
 		return err
 	}
-	defer os.RemoveAll(tmpDir)
-
-	execPath, err := tfinstall.Find(ctx, tfinstall.LookPath(), tfinstall.LatestVersion(tmpDir, false))
+	defer os.RemoveAll(installDir)
```

Comment:
```suggestion
	t.Cleanup(func() {
		if err := os.Remove(installDir); err != nil {
			t.Fatal(err)
		}
	})
```
`defer` is not wrong, but in the context of tests `t.Cleanup` is a bit more flexible especially once/if we add subtests and/or decide to run tests in parallel. It also allows us to fail if some files are left behind which would likely help us discover a bug upstream in hc-install `Remove`.

Reply:
It's not a test function here.

Reply:
Ooops, yes - you're right, this isn't a test 😄  my bad